Q1.java

```diff
@@ -0,0 +1,161 @@
+import java.util.*;
+
+class Q1
+{
+	public static void main(String[] args)
+	{
+		ArrayList<String> words = new ArrayList<String>();
+		words.add("ART");
+		words.add("RAT");
+		words.add("CAT");
+		words.add("CAR");
+
+		printAlphabet(alphabet(words));
+
+	}
+
+
+
+	public static ArrayList<Character> alphabet (ArrayList<String> words)
+	{
+		Map letters = findLetters(words);
+		ArrayList<Node> allNodes = createNodes(letters);
+
+		links(words, allNodes);
+		return getAlphabet(allNodes);
+	}
+
+		
+
+	//getting all of the letters in the language 
+	private static Map findLetters(ArrayList<String> words)
+	{
+		Map letters = new HashMap();
```

Comment:
A few comments here:
- Modern java Maps should always be generified with the type, here it should be Map<Character, Boolean>
- Why are you using a Map? You only seem to use it as a Set? Wouldn't a Set do?
- There is no real benefit doing a contains() before adding here. Most Map/Set data structures the cost of a put() is similar to the cost of a contains(), so by doing the contains check instead of always adding you double your work.
- Modern java usually uses the for(String word : words) style of iteration, unless you really need the loop index.